projects/transformers/experiments/finetuning.py

```diff
@@ -133,39 +194,48 @@
 # where they propose a "simple but hard to beat" approach
 #       https://openreview.net/pdf?id=nzpLWnVAyah
 #
-# How to decided num_train_epochs for each task:
-# They recommend 20 epochs for rte, which is about 50k iterations. They also claim that
-# the number of iterations is more important than dataset size. Here I aim for 50k
-# iterations, unless the size of the training set is already > 50k.
+# How to training time for each task:
+# They recommend 20 epochs for rte, which is about 50k examples. With a batch size
+# of 32, thats about 1562 steps. They also claim that the number of examples is
+# more important than dataset size. Here I aim for 1562 steps unless the size of
+# the training set is already > 50k.
 #
 #       if len(train_dataset) < 50k
 #           train for ~ 50k iterations = round(50k / len(train_dataset))
 #           (cola, mrpc, stsb, rte, wnli)
 #
-#       elif 50k <= len(train_dataset) < 300k
+#       else
 #           use the default of 3 epochs
-#           (sst2, wnli)
 #
-#       elif len(train_dataset) >= 300k
-#           train for 1 epoch
-#           (qqp, mnli)
+# Note that EarlyStoppingCallback is in use, which was not mentioned in the paper
 
 finetuning_bert100k_glue_simple = deepcopy(finetuning_bert100k_glue)
 finetuning_bert100k_glue_simple.update(
-    warmup_ratio=0.1,
+    run_name="finetuning_bert100k_glue_simple",
     task_hyperparams=dict(
-        cola=dict(num_train_epochs=6, num_runs=5),  # 6 * 8500 ~ 50k
+
+        cola=dict(eval_steps=50,
+                  max_steps=1562,
```

Comment:
Could this calculation be donen automatically via a function, given there is a defined heuristic which is given in the comment above? That would avoid having the user calculate it when creating the config. The fewer human interactions the better, not just for efficiency but to avoid accidental data entry mistakes.